app/src/main/java/com/mybus/activity/DisplayFavoritesActivity.java

```diff
@@ -20,4 +110,83 @@ public int getToolbarId() {
     protected int getToolbarTittle() {
         return R.string.displayFavoritesToolbarTittle;
     }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        switch (resultCode) {
+            case RESULT_CANCELED:
+                //The user canceled
+                break;
+            case RESULT_OK:
+                newAddress = data.getStringExtra(SearchActivity.RESULT_STREET_EXTRA);
+                FavoriteNameAlertDialog favoriteNameAlertDialog = new FavoriteNameAlertDialog();
+                switch (requestCode) {
+                    case ADD_SEARCH_RESULT_ID:
+                        favoriteNameAlertDialog.setActionAdding(this);
```

Comment:
Same as above, is there another use of the callback so you separate it into two callbacks?